src/main/java/pl/zimi/zadanka/Zad4.java

```diff
@@ -0,0 +1,94 @@
+package pl.zimi.zadanka;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/*
+source
+https://leetcode.com/problems/reverse-words-in-a-string/
+ */
+public class Zad4 {
+    public static String reverseString(String s) {
+
+        String[] splitedS = s.split(" ");
+        String sReverse = "";
+        boolean flaga = false;
+        for (int i = splitedS.length - 1; i >= 0; i--) {
+
+            if (splitedS[i].length() > 0) {
+//            if (true) {     //ten zostawia artefakty na koncu linii
+                if (flaga == true) {
+                    sReverse += " " + splitedS[i];
+                } else {
+                    sReverse += splitedS[i];
+                    flaga = true;
+                }
+            }
+        }
+        return sReverse;
+    }
+
+    public static String reverseStringPROBLEMY(String s) {
+        List<String> myAlist = new ArrayList<String>();
+        myAlist.add("Geeks");
+        myAlist.add("Practice");
+        myAlist.add("Quiz");
+        System.out.println("Original ArrayList : " + myAlist);
+        myAlist.remove("Quiz");
+        System.out.println("Modified ArrayList : " + myAlist);
+
+
+        String sReverse = new String();
+        List<String> elements = new ArrayList<>();
+        String[] splitedS = s.split(" ");
+//        List <String> bonus = new ArrayList<String>(Arrays.asList(splitedS));
+//        elements.addAll(bonus);
+        elements = Arrays.asList(splitedS);
+
+        String whitespace = "good";
+//        ??? czemu nie działa contains
+//        elements.removeIf(n->n.contains(""));
+        elements.remove("good");
+//        elements.removeAll(Collections.singleton("good"));
+//
+        for (String k : elements) {
+            if (k == "") {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [ES: Comparison of String objects using == or != (ES_COMPARING_STRINGS_WITH_EQ)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq).
This code compares java.lang.String objects for reference equality using the == or != operators. Unless both strings are either constants in a source file, or have been interned using the String.intern() method, the same string value may be represented by two different String objects. Consider using the equals(Object) method instead.

Reply:
Thx a lot! now that's make a sense!

Reply:
Happy to help.